core/src/sumeragi/main_loop.rs

```diff
@@ -909,7 +966,11 @@ pub(crate) fn run<F: FaultInjection>(
                                 BlockCommitted::from(committed_block.clone()),
                             );
 
-                            sumeragi.broadcast_packet(msg, current_topology);
+                            if is_genesis_peer && sumeragi.debug_force_soft_fork {
```

Comment:
this could be guarded by `debug_assertions`?

Reply:
LLVM can see that this condition is always `false` so it won't matter all that much.

Reply:
Sure? It's a small gain for release builds but I'll add it.